src/status_im/components/toolbar_new/view.cljs

```diff
@@ -31,19 +38,9 @@
   (let [style (merge (st/toolbar-wrapper background-color) style)]
     [view {:style style}
      [view st/toolbar
-      [view (st/toolbar-nav-actions-container actions)
-       (when-not hide-nav?
-         (if nav-action
-           [touchable-highlight {:style    st/toolbar-button
-                                 :on-press (:handler nav-action)}
-            [view
-             [image (:image nav-action)]]]
-           [touchable-highlight {:style               st/toolbar-button
-                                 :on-press            #(dispatch [:navigate-back])
-                                 :accessibility-label id/toolbar-back-button}
-            [view
-             [image {:source {:uri :icon_back_dark}
-                     :style  icon-default}]]]))]
+      (when-not hide-nav?
+        [view (st/toolbar-nav-actions-container actions)
+         (create-nav-button (or nav-action act/default-back))])
```

Comment:
why is `create-nav-button` called as function instead of using it as hiccup element?